PlaceRouteHierFlow/cap_placer/capplacer.cpp

```diff
@@ -110,182 +97,165 @@ void Placer_Router_Cap::Placer_Router_Cap_function(vector<int> & ki, vector<pair
 
   vector<string> obs;
 
-  if(lefData.find(unit_capacitor)==lefData.end()){
-  cerr<<"CCCap-Error: unit capacitor not exist in map:"<<" "<<unit_capacitor<<endl;
-  }else{
+  unordered_set<string> obs_map;
 
-  
-  for(unsigned int i=0;i<lefData[unit_capacitor].interMetals.size();i++){
-       int found = 0;
-       for(unsigned int j=0;j<obs.size();j++){
-          if(obs[j]==lefData[unit_capacitor].interMetals[i].metal){
-             found = 1;
-            }
-         }
-       if(found == 0){obs.push_back(lefData[unit_capacitor].interMetals[i].metal);}
+  const auto &uc = lefData.at(unit_capacitor);
+
+  for(unsigned int i=0;i<uc.interMetals.size();i++){
+      obs_map.insert( uc.interMetals[i].metal);
+      if( std::find( obs.begin(), obs.end(), uc.interMetals[i].metal) == obs.end()) {
+	  obs.push_back(uc.interMetals[i].metal);
       }
+  }
+
+  assert( obs_map.size() == obs.size());
+
+  unit_cap_demension.first = uc.width;
+  unit_cap_demension.second= uc.height;
+
+  //round up to grid size
+  auto roundup = []( int& v, int pitch) {
+      v = pitch*((v+pitch-1)/pitch);
+  };
+  //  roundup( unit_cap_demension.first, 80);
```

Comment:
@Lastdayends Can you help me figure out the general way to determine the pitches that we should round the width (and height) too. The value 80 and 84 only work for the Mock PDK.

Reply:
@stevenmburns 

Added. But I don't think roundup function make sense. The width/height of unit cap should be on grid for different PDK, which means unit cap should be updated according to differnt PDK. Besides, this roundup function will result in off grid metal for physical metal assignment.

Yaguang